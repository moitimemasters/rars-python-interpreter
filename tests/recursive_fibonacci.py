def fib_canonical(n):
    if n < 2:
        return 1
    return fib_canonical(n - 1) + fib_canonical(n - 2)


def helper(a, b, x):
    return helper(b, a + b, x - 1) if x else a


def optimized_fib(n):
    return helper(1, 1, n)


optimized_fib(10) - fib_canonical(10)
