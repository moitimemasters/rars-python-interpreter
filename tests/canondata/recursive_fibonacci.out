file: tests/recursive_fibonacci.py
started lexing
ended lexing
started parsing ast
address of if condition: 268767832
ended parsing ast
started compiling
ended compiling
compiled instructions:
	 Instruction: 10
	 Instruction: 8
	 Instruction: 0
	 Instruction: 2
	 Instruction: 7
	 Instruction: 0
	 Instruction: 12
	 Instruction: 0
	 Instruction: 2
	 Instruction: 7
	 Instruction: 0
	 Instruction: 12
	 Instruction: 7
	 Instruction: 11
	 Instruction: 10
	 Instruction: 8
	 Instruction: 10
	 Instruction: 8
	 Instruction: 2
	 Instruction: 0
	 Instruction: 12
	 Instruction: 2
	 Instruction: 0
	 Instruction: 12
	 Instruction: 7
started interpreting
ERROR: Variable not found
